src/services/messengerSync/consumer.js

```diff
@@ -14,31 +14,49 @@ const validateMessage = (content) => {
 		logger.error(`${errorMsg}, action missing.`, content);
 		return false;
 	}
-	if (!content.action || !Object.values(ACTIONS).includes(content.action)) {
-		logger.error(`${errorMsg}, invalid action.`, content);
-		return false;
-	}
 
-	// entity to sync
-	if (content.userId) {
-		if (!ObjectId.isValid(content.userId)) {
-			logger.error(`${errorMsg}, invalid userId.`, content);
-			return false;
+	switch (content.action) {
+		case ACTIONS.SYNC_SCHOOL: {
+			if (!content.schoolId) {
+				logger.error(`${errorMsg}, schoolId is required for ${ACTIONS.SYNC_SCHOOL}.`, content);
+				return false;
+			}
+
+			if (!ObjectId.isValid(content.schoolId)) {
+				logger.error(`${errorMsg}, invalid schoolId.`, content);
+				return false;
+			}
+
+			if (!content.fullSync) {
+				logger.error(`${errorMsg}, fullSync flag has to be set for ${ACTIONS.SYNC_SCHOOL}.`, content);
```

Comment:
Logically this makes no sense as it is just always required here...

Reply:
Yes it is, the flag is only set for readability of the message.